qubessalt/__init__.py

```diff
@@ -174,26 +169,20 @@ def salt_call(self, command='state.highstate', return_output=False):
             qrexec_policy(dispvm.name, self.vm.name, False)
             try:
                 dispvm.force_shutdown()
-            except qubes.qubes.QubesException:
+            except qubes.exc.QubesException:
                 pass
             dispvm.remove_from_disk()
-            self.qc.lock_db_for_writing()
-            self.qc.load()
-            self.qc.pop(dispvm.qid)
-            self.qc.save()
-            self.qc.unlock_db()
+            del self.app.domains[dispvm.name]
```

Comment:
Use `dispvm.cleanup()` to handle all the cleanup, instead of manual one (force_shutdown, remove_from_disk, del).
Also `self.app.save()` isn't necessary here, nor after creating dispvm (see `from_appvm` docstring).